src/test/java/com/palantir/gradle/versions/CheckOverbroadConstraintsTest.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.versions;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.palantir.gradle.versions.lockstate.LockState;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+
+class CheckOverbroadConstraintsTest {
+
+    private static VersionsProps createVersionProps(String... lines) {
+        return VersionsProps.fromLines(List.of(lines), null);
+    }
+
+    private static LockState createLockState(String... productionLines) {
+        ConflictSafeLockFile lockReader = new ConflictSafeLockFile(null);
+        return LockState.from(lockReader.parseLines(Stream.of(productionLines)), lockReader.parseLines(Stream.empty()));
+    }
+
+    @Test
+    void test_no_missing_pins_returns_empty() {
```

Comment:
There's no need to prefix every test method with `test`.

Reply:
sorry - old habbits!